back/engines/free/email_campaigns/app/views/email_campaigns/ideas/_idea-minimal-with-increments.mjml

```diff
@@ -1,6 +1,6 @@
 <tr style="border: 1px solid #ecedee; background-color: #FFFFFF" class="idea">
   <td width="200" style="text-align: left;">
-    <a style="font-size: 14px; font-weight: 300; color: #000; text-decoration: none;" href="{{top_ideas.url}}">
+    <a style="font-size: 14px; font-weight: 300; color: #000; text-decoration: none;" href="<%= serialized_idea.url %>">
```

Comment:
@nTraum What is the `serialized_idea`? I can only find this in Ruby in the `after_destroy` method in `SideFxIdeaService`.

Reply:
@jinjagit good question!

The `Campaigns::AdminDigest` class is responsible for collecting and transforming all the data we want to use in the email we are sending out to admins. In particular `Campaigns::AdminDigest#new_initiatives` collects them:

```ruby
    def new_initiatives(time: Time.zone.today)
      @new_initiatives ||= Initiative.published.where('published_at > ?', (time - 1.week))
                                     .order(published_at: :desc)
                                     .includes(:initiative_images)
                                     .map(&method(:serialize_initiative))
    end

```
https://github.com/citizenlabdotco/citizenlab/blob/CL2-6824_broken_links_admin_digest/back/engines/free/email_campaigns/app/models/email_campaigns/campaigns/admin_digest.rb#L203-L208


The file I have modified is a partial and part of the admin digest email we want to render.
The email contains multiple sections, one of them being the "new initiatives" that is rendered [here](https://github.com/citizenlabdotco/citizenlab/blob/CL2-6824_broken_links_admin_digest/back/engines/free/email_campaigns/app/views/email_campaigns/admin_digest_mailer/campaign_mail.mjml#L4).

The `new_iniatives` partial iterates over each `new_iniative`:

```erb
<mj-raw><% event.new_initiatives.each do |serialized_initiative| %></mj-raw>
    <%= render 'email_campaigns/initiatives/initiative', serialized_initiative: serialized_initiative %>
<mj-raw><% end %></mj-raw>
```

https://github.com/citizenlabdotco/citizenlab/blob/CL2-6824_broken_links_admin_digest/back/engines/free/email_campaigns/app/views/email_campaigns/initiatives/_new-initiatives.mjml#L9-L11

This is where the variable `serialized_idea` comes from. Does that clear it up?

Reply:
Ah yes, very helpful. Thanks!